python/pyarrow/tensor.pxi

```diff
@@ -595,7 +595,25 @@ shape: {0.shape}""".format(self)
 
 cdef class SparseCSRMatrix(_Weakrefable):
     """
-    A sparse CSR matrix.
+    SparseCSRMatrix represents a sparse matrix in Compressed Sparse Row (CSR) format.
+
+    Attributes:
+        indptr : array
+            Index pointer array.
+        indices : array
+            Column indices of the corresponding non-zero values.
+        shape : tuple
+            Shape of the matrix.
+        dim_names : list, optional
+            Names of the dimensions.
+
+    Example:
+        >>> import pyarrow as pa
+        >>> indptr = pa.array([0, 2, 3])
+        >>> indices = pa.array([0, 2, 1])
+        >>> shape = (2, 3)
+        >>> tensor = pa.SparseCSRMatrix(indptr, indices, shape)
+        >>> print(tensor)
```

Comment:
Did you meant to use one of the `from_*` methods here?

We do not use `SparseCSRMatrix`'s constructor directly, see: https://github.com/apache/arrow/blob/ada8750917cd2ec2ae8a605d2673d78ef04c82e4/python/pyarrow/tensor.pxi#L617-L619

Could you change the example to use one of `pyarrow.SparseCSRMatrix.from_*` functions and also add examples in separate `pyarrow.SparseCSRMatrix.from_*` functions?

Attributes part should also be removed here (note we use NumPy style docstrings).

Could we also then add similar examples (with outputs) to other tensor formats/classes?